packages/admin-panel/src/pages/ProfilePage.js

```diff
@@ -35,41 +48,118 @@ const SuccessMessage = styled(SmallAlert)`
   margin-bottom: 1.5rem;
 `;
 
+const STATUS = {
+  IDLE: 'idle',
+  LOADING: 'loading',
+  SUCCESS: 'success',
+  ERROR: 'error',
+  DISABLED: 'disabled',
+};
+
+const initialState = {
+  status: STATUS.IDLE,
+  errorMessage: null,
+  successMessage: null,
+  confirmModalIsOpen: false,
+  profileImage: {
+    fileId: null,
+    data: null,
+  },
+};
+
+const initReducer = user => {
+  return { ...initialState, profileImage: { data: user.profileImage, fileId: null } };
+};
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'loading': {
+      return {
+        ...state,
+        status: STATUS.LOADING,
+        errorMessage: null,
+        successMessage: null,
+      };
+    }
+    case 'fileChange': {
+      return {
+        ...state,
+        profileImage: action.payload,
+        status: STATUS.IDLE,
+      };
+    }
+    case 'deleteProfile': {
+      return { ...state, profileImage: { fileId: '-', data: null }, confirmModalIsOpen: false };
```

Comment:
We need to set a file id to something so that the backend will pick up the profileImage field as an updatedField and save the change.